### PR TITLE
fix: Add typing for specType field on spec method

### PR DESCRIPTION
**Pull request by NickBolles**
* Closes 

### User facing changelog
Add typings for `specType` field on `Cypress.spec`

### Additional details

Typescript users will get a compilation error

### PR Tasks

* [ ] Have tests been added/updated?
**n/a**
* [x] Has the original issue been tagged with a release in ZenHub? 
**n/a**
* [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
**n/a**
* [x] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
* [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?
**n/a**

┆Issue is synchronized with this [Jira Features](https://cypress-io.atlassian.net/browse/TR-234) by [Unito](https://www.unito.io/learn-more)

### CODE REVIEW

**Comment by NickBolles** [comment]
@jennifer-shehane I just tried something, but I'm on my phone. If that doesn't work I'll check it out tomorrow on my computer.

**Comment by jennifer-shehane** [comment]
@NickBolles Do you still have time to address the changes requested in this PR? We will have to close this PR due to inactivity otherwise.

**Comment by NickBolles** [comment]
@jennifer-shehane That should take care of the linting issues. Thanks for the ping, let me know if there's anything else I need to do.

File: cli/types/cypress.d.ts [hunk 1]
```diff
@@ -107,6 +107,14 @@ declare namespace Cypress {
     fromAutWindow: WindowPosition & { x: number, y: number }
   }
 
+  /**
+   * Spec type for the given test. "integration" is the default, but
+   * test run with the experimentalComponentTesting tests will be "component"
+   *
+   * @see https://docs.cypress.io/guides/references/experiments.html#Component-Testing
+   */
```
Comment (jennifer-shehane) [inline cli/types/cypress.d.ts:115]:
```suggestion
  /**
   * Spec type for the given test. "integration" is the default, but
   * tests run using experimentalComponentTesting will be "component"
   *
   * @see https://on.cypress.io/experiments
   */
```

**Review by jennifer-shehane — DISMISSED** [review]

File: cli/types/cypress.d.ts [hunk 2]
```diff
@@ -229,6 +238,7 @@ declare namespace Cypress {
       relative: string // "cypress/integration/config_passing_spec.coffee" or "__all" if clicked all specs button
       absolute: string
       specFilter?: string // optional spec filter used by the user
+      specType?: CypressSpecType
```
Comment (jennifer-shehane) [inline cli/types/cypress.d.ts:241]: This is not optional

```suggestion
      specType: CypressSpecType // "integration" is the default
```

File: cli/types/cypress.d.ts [hunk 3]
```diff
@@ -229,6 +238,7 @@ declare namespace Cypress {
       relative: string // "cypress/integration/config_passing_spec.coffee" or "__all" if clicked all specs button
       absolute: string
       specFilter?: string // optional spec filter used by the user
+      specType: CypressSpecType // "integration" is the default
```
Comment (jennifer-shehane) [inline cli/types/cypress.d.ts:241]:
```suggestion
      specType: CypressSpecType
```

**Review by jennifer-shehane — APPROVED** [review]

File: cli/types/cypress.d.ts [hunk 4]
```diff
@@ -229,6 +238,7 @@ declare namespace Cypress {
       relative: string // "cypress/integration/config_passing_spec.coffee" or "__all" if clicked all specs button
       absolute: string
       specFilter?: string // optional spec filter used by the user
+      specType: CypressSpecType
```
Comment (jennifer-shehane) [inline cli/types/cypress.d.ts:241]:
```suggestion
      specType?: CypressSpecType
```

**Review by jennifer-shehane — APPROVED** [review]

**Pull request merged**